doc/langdef.md

```diff
@@ -116,44 +126,185 @@ In general it is a bad idea for those defining contexts or extensions to use
 identifiers that are reserved words in programming languages which might embed
 CEL.
 
+### Name Resolution
+
+A CEL expression is parsed in the scope of a specific protocol buffer package,
+which controls the interpretation of names. The scope is set by the application
+context of an expression. A CEL expression can contain simple names as in `a` or
+qualified names as in `a.b`. The meaning of such expressions is a combination of
+one or more of:
+
+*   Variables and Functions: some simple names refer to variables in the
+    execution context, standard functions, or other name bindings provided by
+    the CEL application.
+*   Field selection: appending a period and identifier to an expression could
+    indicate that we're accessing a field within a protocol buffer or map.
```

Comment:
Phrasing nit.
```
Field selection: a dot-delimited expression could indicate field access
within a value, such as a protocol buffer or map
```
Any kind of expression, not just identifiers, can be the operand of a field selection.

Reply:
Correct - and the original text says so.  I don't understand "dot-delimited".  Is that the production
"." Expr "."
perhaps?  I don't see the replacement as a big improvement.